### PR TITLE
Use shorthand for constructor properties

### PR DESCRIPTION
**Pull request by zachkont**
**Description**:

Use shorthand for constructor properties

**Checklist**:

- [x] Tested the bot on both clients with a few commands
- [x] Updated the [`README`](README.md) if necessary
- [x] Updated the version string in the [`package.json`](package.json) if necessary

### CODE REVIEW

File: src/game.ts [hunk 1]
```diff
@@ -16,44 +16,26 @@ export type Providers = {
...
-   * @param  {string} label - The human-formatted label of the game.
+  /**
+   * Creates an instance of Game.
+   * @param {string} name The internal name of the game.
+   * @param {string[]} aliases The aliases the game uses.
+   * @param {string} label The human-formatted label of the game.
+   * @param {string} color The color representing the game.
+   * @param {string} icon The game icon.
+   * @param {Providers} providers The game providers.
+   * @param {TelegramIVTemplate[]} telegramIVTemplates The Telegram IV templates.
```
Comment (TimJentzsch) [inline src/game.ts:27]: I don't think we need explicit types here, they are derived from the constructor types

File: src/notifications/notification.ts [hunk 2]
```diff
@@ -5,47 +5,30 @@ import { StrUtil } from '../util/util';
...
+   * Creates a new Notification.
+   * @param {Date} timestamp The timestamp of the notification.
+   * @param {Game} game The game the notification is for.
+   * @param {NotificationElement} title The title of the notification.
+   * @param {string} [content] The content of the notification.
+   * @param {NotificationElement} [author] The author of the notification.
+   * @param {string} [color] The color of the notification.
+   * @param {string} [thumbnail] The (small) thumbnail of the notification.
+   * @param {string} [image] The (big) image of the notification.
+   * @param {NotificationElement} [footer] The footer of the notification.
```
Comment (TimJentzsch) [inline src/notifications/notification.ts:18]: I think we can remove the types here

File: src/notifications/notification_element.ts [hunk 3]
```diff
@@ -1,11 +1,3 @@
...
-  public text: string;
-  public link?: string;
-  public icon?: string;
-
-  constructor(text: string, link?: string, icon?: string) {
-    this.text = text;
-    this.link = link;
-    this.icon = icon;
-  }
+  constructor(public text: string, public link?: string, public icon?: string) {}
```
Comment (TimJentzsch) [inline src/notifications/notification_element.ts:2]: We will need comments for that at some point

File: src/providers/provider.ts [hunk 4]
```diff
@@ -3,15 +3,9 @@ import Notification from '../notifications/notification';
...
-  public url: string;
-  public label: string;
-  public game: Game;
   public logger: Logger;
 
-  constructor(url: string, label: string, game: Game) {
-    this.url = url;
-    this.label = label;
-    this.game = game;
+  constructor(public url: string, public label: string, public game: Game) {
```
Comment (TimJentzsch) [inline src/providers/provider.ts:8]: We'll need comments for this at some point

File: src/reddit/reddit_user.ts [hunk 5]
```diff
@@ -1,12 +1,10 @@
 import Logger from '../logger';
 
 export default class RedditUserProvider {
-  public name: string;
   public titleFilter: RegExp;
   public static logger = new Logger('Reddit User');
 
-  constructor(name: string, titleFilter?: string | RegExp) {
-    this.name = name;
+  constructor(public name: string, titleFilter?: string | RegExp) {
```
Comment (TimJentzsch) [inline src/reddit/reddit_user.ts:7]: We'll need comments for this at some point

File: src/rss/rss_item.ts [hunk 6]
```diff
@@ -1,28 +1,14 @@
...
-    this.timestamp = timestamp;
-    this.feed = feed;
-  }
+    public title: string,
+    public author: string,
+    public link: string,
+    public content: string,
+    public timestamp: Date,
+    public feed: { name?: string; source?: string; link?: string },
+  ) {}
```
Comment (TimJentzsch) [inline src/rss/rss_item.ts:11]: We'll need comments for this at some point

File: src/updater.ts [hunk 7]
```diff
@@ -11,28 +11,31 @@ export default class Updater {
...
-  /** Creates a new Updater.
-   * @param {number} cyleInterval - The initial delay in seconds.
+  /**
+   * Creates an instance of Updater.
+   * @param {string} key
+   * @param {boolean} enabled
+   * @param {boolean} autosave
+   * @param {number} limit
+   * @param {number} gameInterval
+   * @param {number} cycleInterval
```
Comment (TimJentzsch) [inline src/updater.ts:29]: I don't think we need the types here

File: src/user.ts [hunk 8]
```diff
@@ -15,15 +15,13 @@ export enum UserRole {
...
-  public bot: BotClient;
-  /** The unique ID of the user. */
-  public id: string;
-  /** Create a new BotUser.
-   *
-   * @param id - The ID of the BotUser.
+  /**
+   * Create a new BotUser.
+   * @param {BotClient} bot The bot the user is associated to.
+   * @param {string} id The ID of the BotUser.
```
Comment (TimJentzsch) [inline src/user.ts:21]: I don't think we need the types here

File: src/game.ts [hunk 9]
```diff
@@ -16,44 +16,26 @@ export type Providers = {
...
+   * @memberof Game
```
Comment (TimJentzsch) [inline src/game.ts:28]: What does this do?

**Review by TimJentzsch — CHANGES_REQUESTED** [review]

File: src/game.ts (hunk 9 above)
Comment (zachkont) [reply src/game.ts]: removed

File: src/channel.ts [hunk 10]
```diff
@@ -149,22 +135,23 @@ export default class Channel {
...
 
-  /** Creates a new Channel. */
+  /**
+   * Creates an instance of Channel.
+   * @param {string} id The unique ID of the channel
+   * @param {BotClient} bot The BotClient this channel is used in.
+   * @param {Game[]} [gameSubs] The games this channel is subscribed to.
+   * @param {string} [prefix] The prefix the channel uses.
+   * @param {string} [label] The label of the channel (if specified).
+   * @param {boolean} [disabled=false] Disabled channels won't receive automatic updates.
```
Comment (zachkont) [inline src/channel.ts:145]:
```suggestion
   * @param id The unique ID of the channel
   * @param bot The BotClient this channel is used in.
   * @param gameSubs The games this channel is subscribed to.
   * @param prefix The prefix the channel uses.
   * @param label The label of the channel (if specified).
   * @param disabled Disabled channels won't receive automatic updates.
```

File: src/channel.ts [hunk 11]
```diff
@@ -149,22 +135,23 @@ export default class Channel {
...
+   */
```
Comment (TimJentzsch) [inline src/channel.ts:146]:
```suggestion
  /**
   * Creates an instance of Channel.
   * @param id The unique ID of the channel
   * @param bot The BotClient this channel is used in.
   * @param gameSubs The games this channel is subscribed to.
   * @param prefix The prefix the channel uses.
   * @param label The label of the channel (if specified).
   * @param disabled Disabled channels won't receive automatic updates.
   */
```

**Review by TimJentzsch — CHANGES_REQUESTED** [review]

**Review by TimJentzsch — APPROVED** [review]

**Pull request merged**